lib/ide/src/controller/project.rs

```diff
@@ -62,7 +61,25 @@ shared! { Handle
     }
 }
 
+impl State {
+    /// Creates a new project controller. Schedules all necessary execution with
+    /// the global executor.
+    pub fn new_running(file_manager_transport:impl Transport + 'static) -> Self {
```

Comment:
I'm not sure if this should be a public function. Or just merge with these function in `Handle` The docs say that this function "creates a new project controller", but in fact this create _State_ and state alone is not project controller, is it?